src/bun.js/bindings/webcore/JSMIMEParams.cpp

```diff
@@ -0,0 +1,732 @@
+    // Set [Symbol.iterator] to entries
+    putDirectWithoutTransition(vm, vm.propertyNames->iteratorSymbol, getDirect(vm, Identifier::fromString(vm, "entries"_s)), PropertyAttribute::DontEnum | 0);
+
+    // Set toJSON to toString
+    putDirectWithoutTransition(vm, vm.propertyNames->toJSON, getDirect(vm, vm.propertyNames->toString), PropertyAttribute::Function | 0);
+
+    JSC_TO_STRING_TAG_WITHOUT_TRANSITION();
+}
+
+//-- JSMIMEParamsConstructor Implementation --
+
+const ClassInfo JSMIMEParamsConstructor::s_info = { "MIMEParams"_s, &Base::s_info, nullptr, nullptr, CREATE_METHOD_TABLE(JSMIMEParamsConstructor) };
+
+JSMIMEParamsConstructor* JSMIMEParamsConstructor::create(VM& vm, Structure* structure, JSObject* prototype)
+{
+    JSMIMEParamsConstructor* constructor = new (NotNull, JSC::allocateCell<JSMIMEParamsConstructor>(vm)) JSMIMEParamsConstructor(vm, structure);
+    constructor->finishCreation(vm, prototype);
+    return constructor;
+}
+
+Structure* JSMIMEParamsConstructor::createStructure(VM& vm, JSGlobalObject* globalObject, JSValue prototype)
+{
+    return Structure::create(vm, globalObject, prototype, TypeInfo(InternalFunctionType, StructureFlags), info());
+}
+
+JSMIMEParamsConstructor::JSMIMEParamsConstructor(VM& vm, Structure* structure)
+    : Base(vm, structure, callMIMEParams, constructMIMEParams)
+{
+}
+
+JSC_DEFINE_HOST_FUNCTION(callMIMEParams, (JSGlobalObject * globalObject, CallFrame* callFrame))
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+    return throwVMError(globalObject, scope, createNotAConstructorError(globalObject, callFrame->jsCallee()));
+}
+
+JSC_DEFINE_HOST_FUNCTION(constructMIMEParams, (JSGlobalObject * globalObject, CallFrame* callFrame))
+{
+
+    JSC::VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    auto* zigGlobalObject = defaultGlobalObject(globalObject);
+    JSC::Structure* structure = zigGlobalObject->m_JSMIMEParamsClassStructure.get(zigGlobalObject);
+
+    JSC::JSValue newTarget = callFrame->newTarget();
+    if (UNLIKELY(zigGlobalObject->m_JSMIMEParamsClassStructure.constructor(zigGlobalObject) != newTarget)) {
+        if (!newTarget) {
+            throwTypeError(globalObject, scope, "Class constructor MIMEParams cannot be invoked without 'new'"_s);
+            return {};
+        }
+
+        auto* functionGlobalObject = defaultGlobalObject(getFunctionRealm(globalObject, newTarget.getObject()));
+        RETURN_IF_EXCEPTION(scope, {});
+        structure = JSC::InternalFunction::createSubclassStructure(globalObject, newTarget.getObject(), functionGlobalObject->m_JSMIMEParamsClassStructure.get(functionGlobalObject));
+        RETURN_IF_EXCEPTION(scope, {});
+    }
+
+    // Create the internal JSMap
+    JSMap* map = JSMap::create(vm, globalObject->mapStructure());
+    RETURN_IF_EXCEPTION(scope, {}); // OOM check
+
+    // Create the JSMIMEParams instance
+    JSMIMEParams* instance = JSMIMEParams::create(vm, structure, map);
```

Comment:
does this need a guard too?

Reply:
never makes a throw scope that I can see